fix: Update PR template checklist

ChiefHolland opened the pull request:
# Contributor Comments

This change is to add a requirement for updating documentation when adding, changing, or removing features in a project.

## Pull Request Checklist

Thank you for submitting a contribution to our project!

In order to streamline the review of your contribution we ask that you review
and comply with the below requirements:

- [x] Rebase your branch against the latest commit of the target branch
- [x] If you are adding a dependency, please explain how it was chosen
- [x] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results
- [x] If there is an issue associated with your Pull Request, link the issue to the PR.

JonZeolla committed:
Adding docs update requirement to checklist

ChiefHolland commented on PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,15 +1,14 @@
 # Contributor Comments
 
-
-
```
Removed lines as per markdownlint

ChiefHolland commented on PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,15 +1,14 @@
-- [ ] Rebase your branch against the latest commit of the target branch
-- [ ] If you are adding a dependency, please explain how it was chosen
-- [ ] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results
+- [ ] Rebase your branch against the latest commit of the target branch.
```
Added periods to sentences for consistency

ChiefHolland commented on PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,15 +1,14 @@
+- [ ] If you are adding a dependency, please explain how it was chosen.
+- [ ] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results.
 - [ ] If there is an issue associated with your Pull Request, link the issue to the PR.
+- [ ] If a feature is being added, removed, or changed, code changes must be accompanied by the appropriate documentation updates or additions.
```
Added new PR requirement

JonZeolla replied:
Instead of the `If` approach, what if we just describe the desired outcome?

```suggestion
- [ ] Validate that documentation is accurate and aligned to any project updates or additions.
```

ChiefHolland replied:
I was leaning into consistent sentence structure, but this is cleaner and more to the point. I like it.

JonZeolla committed:
Adding contributor verbiage

ChiefHolland commented on PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,5 +1,7 @@
 # Contributor Comments
 
+[Please insert a description of the change here]
```
Added the contributor verbiage from the CookieCutter template.

JonZeolla replied:
It seems like the team doesn't always remember to replace this.  Is there a better way to do this than [Please insert...]?

What about:
```suggestion
[INSERT DESCRIPTION]
```

ChiefHolland committed:
Update PULL_REQUEST_TEMPLATE.md

Co-authored-by: JonZeolla <Zeolla@gmail.com>

ChiefHolland committed:
Update PULL_REQUEST_TEMPLATE.md

The all caps might work to draw attention to the necessary substitution. I'll address this in cookiecutter, too.

Co-authored-by: JonZeolla <Zeolla@gmail.com>

JonZeolla reviewed (approved):
LGTM
